### PR TITLE
4493 closed cases backend

### PR DESCRIPTION
**Pull request by mrinsin**
closed cases now being successfully retrieved and displayed.

WIP on UI when there are no closed/open cases.

### CODE REVIEW

File: shared/src/business/useCases/getOpenConsolidatedCasesInteractor.test.js [hunk 1]
```diff
@@ -46,15 +48,19 @@ describe('getOpenCasesInteractor', () => {
...
     });
 
     expect(result).toEqual([]);
   });
 
   it('should return a list of open cases', async () => {
-    const result = await getOpenCasesInteractor({
+    applicationContext
+      .getPersistenceGateway()
+      .getOpenCasesByUser.mockResolvedValue([MOCK_CASE]);
```
Comment (kkoskelin) [inline shared/src/business/useCases/getOpenConsolidatedCasesInteractor.test.js]: I think this mock is already set up in the 'beforeEach' and the test will pass without this block.

File: shared/src/persistence/dynamo/cases/getClosedCasesByUser.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * @param {object} providers.applicationContext the application context
+ * @param {string} providers.userId the user id to get closed cases by
+ * @returns {object} the closed cases
+ */
+exports.getClosedCasesByUser = async ({ applicationContext, userId }) => {
+  const userCases = await getCasesByUser({ applicationContext, userId });
+
+  const closedCases = userCases.filter(
+    x => x.status === Case.STATUS_TYPES.closed,
+  );
```
Comment (kkoskelin) [inline shared/src/persistence/dynamo/cases/getClosedCasesByUser.js:17]: I feel like perhaps the query itself (to persistence) ought to contain this condition.  Probably the file `getCasesByUser` could contain another function that shares some query code against Elasticsearch? I'm open to discussion on this -- the query itself may become not-insignificant in terms of number of rows returned.

**Review by kkoskelin — APPROVED** [review]
I'm totally open to being wrong in my opinions. Just say so. :D

File: shared/src/business/useCases/getOpenConsolidatedCasesInteractor.test.js (hunk 1 above)
Comment (mrinsin) [reply shared/src/business/useCases/getOpenConsolidatedCasesInteractor.test.js]: Thank you!!

File: shared/src/persistence/dynamo/cases/getClosedCasesByUser.js (hunk 2 above)
Comment (mrinsin) [reply shared/src/persistence/dynamo/cases/getClosedCasesByUser.js:17]: Ooh, thank you for the suggestion. I would love to pair with you to learn how to add it to the elastic search query!

**Pull request merged**